sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1458,86 +1467,183 @@ static void _process_receive_stats(gnrc_netif_t *netdev, gnrc_pktsnip_t *pkt)
 }
 
 /**
- * @brief   Retrieve the netif event queue if enabled
+ * @brief   Handle failed transmission attempt
  *
- * @param[in]   netif   gnrc_netif instance to operate on
+ * @param   netif       Interface which failed to transfer a frame
+ * @param   error_code  Return value of send() or send_confirm() indicating
+ *                      the failure cause
+ * @param   pkt         Frame that couldn't be send
+ * @param   push_back   Where to enqueue the frame (with module gnrc_netif_pktq)
+ */
+static void _tx_failed(gnrc_netif_t *netif, int error_code, gnrc_pktsnip_t *pkt,
+                       bool push_back)
+{
+    /* arguments may be unused, depending on used pseudo modules */
+    (void)netif; (void)error_code; (void)pkt; (void)push_back;
+    DEBUG("gnrc_netif: error sending packet %p (code: %i)\n",
+          (void *)pkt, error_code);
+    if (IS_USED(MODULE_NETSTATS_NEIGHBOR)) {
+        int8_t retries = -1;
+        netstats_nb_result_t result;
+        if (error_code == -EBUSY) {
+            result = NETSTATS_NB_BUSY;
+        } else {
+            result = NETSTATS_NB_NOACK;
+            netdev_t *dev = netif->dev;
+            dev->driver->get(dev, NETOPT_TX_RETRIES_NEEDED, &retries, sizeof(retries));
+        }
```

Comment:
```suggestion
        switch (error_code) {
        case -EBUSY:
            result = NETSTATS_NB_BUSY;
            break;
        case -ECOMM:
            result = NETSTATS_NB_NOACK;
            {
                netdev_t *dev = netif->dev;
                dev->driver->get(dev, NETOPT_TX_RETRIES_NEEDED, &retries, sizeof(retries));
            }
            break;
        default:
            result = NETSTATS_NB_ERROR;
            break;
        }
```

IMO it would make sense to add at least one more possible result here. According to the API doc, `-ECOMM` is to be used to indicate missing ACK. All remaining error codes could get a catch all error result.

Reply:
There is no driver that uses `-ECOMM` to indicate no-ACK